### DISCUSSION TITLE
How to get the font_scale on the Lua script side.

### DISCUSSION THREAD

**Discussion by daregada** [wezterm/wezterm#3410] [Q&A]
Is there any way to get the `font_scale` in the window on the Lua script side?

What I want to know is the `font_scale` that is scaled by `wezterm.action.IncreaseFontSize` and `DecreaseFontSize`, `RestoreFontSize`.

From a quick look at the source code, I can't find any way to get it on the Lua script side.

**Comment by yutkat** [comment]
Here is the code for font size calculation.
https://github.com/wez/wezterm/blob/78a447768896e8958b3485ae5941eed606f86a3b/wezterm-gui/src/termwindow/resize.rs#L404-L406

I thought you could get font_size with this function, but it doesn't seem to change dynamically.
https://wezfurlong.org/wezterm/config/lua/window/effective_config.html

**Comment by wez** [comment]
Can I ask what you plan to do with the scale? That may influence how the information ultimately gets exposed to lua

**Reply by daregada** [reply]
The reason I want to get the font_scale is because I want to display the font_scale in the window title (for novice users).

Currently I am using my own events and font_scale instead of using IncreaseFontSize or DecreaseFontSize to directly change the overridden font_size.

```lua
local default_font_size = config.font_size
local my_font_scale = 1.0

wezterm.on('decrease-font-size', function(window, pane)
  local overrides = window:get_config_overrides() or {}
  my_font_scale = my_font_scale >= 0.5 and my_font_scale / 1.1 or my_font_scale
  overrides.font_size = default_font_size * my_font_scale
  window:set_config_overrides(overrides)
end)

wezterm.on('increase-font-size', function(window, pane)
  local overrides = window:get_config_overrides() or {}
  my_font_scale = my_font_scale <= 2.0 and my_font_scale * 1.1 or my_font_scale
  overrides.font_size = default_font_size * my_font_scale
  window:set_config_overrides(overrides)
end)

wezterm.on('restore-font-size', function(window, pane)
  local overrides = window:get_config_overrides() or {}
  my_font_scale = 1.0
  overrides.font_size = default_font_size
  window:set_config_overrides(overrides)
end)
```
